File: src/worker/java/MySOWorker.java
import parspice.sender.DoubleArraySender;
import parspice.sender.LongSender;
import parspice.worker.SOWorker;

public class MySOWorker extends SOWorker<Long, double[]> {

    public MySOWorker() {
        super(new LongSender(), new DoubleArraySender(3));
    }

    @Override
    public void setup(Long input) throws Exception {
        // do something with input
    }

    @Override
    public double[] task(int i) throws Exception {
        return new double[]{1., 2., i};
    }
}

File: src/worker/kotlin/MySOWorker.kt
import parspice.sender.DoubleArraySender
import parspice.sender.LongSender
import parspice.worker.SOWorker

class MySOWorker: SOWorker<Long, DoubleArray>(LongSender(), DoubleArraySender(3)) {
    override fun setup(input: Long) {}

    override fun task(i: Int) = doubleArrayOf(1.0, 2.0, i.toDouble())
}